js/ad_manager_vast.js

```diff
@@ -2299,22 +2308,25 @@ OO.Ads.manager(function(_, $) {
      * @method Vast#_beginVpaidAd
      */
     var _beginVpaidAd = _.bind(function() {
-      var ad = currentAd.vpaidAd;
-      //TODO: Is this used for anything?
-      var adLinear = _safeFunctionCall(ad, "getAdLinear");
-
-      initSkipAdOffset(currentAd);
-      //Since a VPAID 2.0 ad handles its own UI, we want the video player to hide its UI elements
-      this.amc.hidePlayerUi();
-      this.amc.notifyPodStarted(currentAd.id, currentAd.ad.adPodLength);
-
-      this.amc.notifyLinearAdStarted(currentAd.id, {
-        name: currentAd.data.title,
-        duration : _safeFunctionCall(ad, "getAdDuration"),
-        clickUrl: currentAd.ad.data.nonLinear.nonLinearClickThrough.length > 0,
-        indexInPod: currentAd.ad.sequence,
-        skippable : _safeFunctionCall(ad, "getAdSkippableState")
-      });
+      if (_isVpaidAd(currentAd)) {
+        var ad = currentAd.vpaidAd;
+        var clickthru = currentAd.ad.data.nonLinear ? currentAd.ad.data.nonLinear.nonLinearClickThrough : '';
```

Comment:
can you do the truthy checks for all these variables.
(currentAd && currentAd.ad && currentAd.ad.data)